pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -89,12 +101,73 @@ func (pc *PackageControllerClient) InstallController(ctx context.Context) error
 	if err = pc.CreateCronJob(ctx); err != nil {
 		logger.Info("Warning: not able to trigger cron job, please be aware this will prevent the package controller from installing curated packages.")
 	}
+
+	if err := pc.waitForActiveBundle(ctx); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// packageBundleControllerResource is the name of the package bundle controller
+// resource in the API.
+const packageBundleControllerResource string = "packageBundleController"
+
+// waitForActiveBundle polls the package bundle controller for its active bundle.
+//
+// It returns nil on success. Success is defined as receiving a valid package
+// bundle controller from the API with a non-empty active bundle.
+//
+// If no timeout is specified, a default of 1 minute is used.
+func (pc *PackageControllerClient) waitForActiveBundle(ctx context.Context) error {
+	timeout := time.Minute
+	if pc.activeBundleTimeout > 0 {
+		timeout = pc.activeBundleTimeout
+	}
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	done := make(chan error)
+	go func() {
+		defer close(done)
+		pbc := &packagesv1.PackageBundleController{}
+		for {
+			err := pc.kubectl.GetObject(timeoutCtx, packageBundleControllerResource, pc.clusterName,
+				packagesv1.PackageNamespace, pc.kubeConfig, pbc)
+			if err != nil {
+				done <- fmt.Errorf("getting package bundle controller: %w", err)
+				return
+			}
+
+			if pbc.Spec.ActiveBundle != "" {
+				logger.V(6).Info("found packages bundle controller active bundle",
+					"name", pbc.Spec.ActiveBundle)
+				return
+			}
+
+			logger.V(6).Info("waiting for package bundle controller to activate a bundle",
```

Comment:
Do we want to add a retry counter that way its easier to follow in the log to figure out how much time has passed?

Reply:
Could... But log messages are timestamped, no?